src/core/gui/XournalppCursor.cpp

```diff
@@ -515,6 +521,71 @@ auto XournalppCursor::createHighlighterOrPenCursor(int size, double alpha) -> Gd
 }
 
 
+auto XournalppCursor::createHorizontalLineCursor() -> GdkCursor* {
+    GtkWidget* theWidget = control->getWindow()->getXournal()->getWidget();
+    GdkWindow* theWindow = gtk_widget_get_window(theWidget);
+
+    gint x, y;
+    GdkDevice *mouse_device;
+
+    #if GTK_CHECK_VERSION (3,20,0)
+    GdkSeat *seat = gdk_display_get_default_seat (gdk_display_get_default ());
+    mouse_device = gdk_seat_get_pointer (seat);
+    #else
+    GdkDeviceManager *devman = gdk_display_get_device_manager (gdk_display_get_default ());
+    mouse_device = gdk_device_manager_get_client_pointer (devman);
+    #endif
+
+    gdk_window_get_device_position (theWindow, mouse_device, &x, &y, nullptr);
+
+    Layout* const layout = this->control->getWindow()->getLayout();
+    XojPageView* page = layout->getPageViewAt(x, y);
+    if (page) {
+        Range range = page->getVisiblePart();
+        double zoom = control->getZoomControl()->getZoom();
+        const int width = static_cast<int>(std::round(range.getWidth() * zoom));
+        const int height = 5; // Five is the default cursor size used elsewhere in the app.
+        gint dx = x - static_cast<int>(range.getX() * zoom) - page->getX();
+        // 64-bit field containing the current value of dx and width to check
+        // if we need to re-draw the cursor
+        auto flavour = (gulong) (dx << (sizeof(gint) * 4) | width);
```

Comment:
This assumes that width is 32 bits, dx is 32 bits, and flavour is 64 bits. I would make this explicit by assigning `int32_t` to width and dx, and `int64_t` to flavour.

Reply:
Also, it's not very clear to me whether dx and width are operated over as 64 bit integers in that last expression. Do they need to be cast to a 64 bit integer first?

Reply:
The `auto` keyword should inherit from the cast (which makes the bit shift explicit), but I can re-assign width and dx.

Reply:
Sorry, I didn't explain my concern very clearly. See my new review comment.